File: micropython/Sensorkit/ReedSwitch.py
from machine import Pin
# a Reed switch sensor
class ReedSwitch():
  def __init__(self, SWITCH):
    global pinSwitch
    pinSwitch = Pin(SWITCH, Pin.IN, Pin.PULL_UP)
  def open(self):
    return pinSwitch.value()
  def closed(self):
    return 1-self.open()

File: arduino/Sensorkit/src/ReedSwitch.cpp
#include "ReedSwitch.h"

ReedSwitch::ReedSwitch(int pin) { 
  pinSwitch = pin; 
}
void ReedSwitch::setup() {
  pinMode(pinSwitch, INPUT_PULLUP);
}
int ReedSwitch::open() { 
  return digitalRead(pinSwitch); 
}
int ReedSwitch::closed() {
  return 1 - open(); 
}
